### PR TITLE
g3log: add version 2.1

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **g3log/***

- add version 2.1
- add C++17 mode for g3log>=2.0
- use `add_subdirectory` in test_v1_package

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

File: recipes/g3log/all/conanfile.py [hunk 1]
```diff
@@ -82,9 +95,7 @@ def loose_lt_semver(v1, v2):
 
         minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
         if minimum_version and loose_lt_semver(str(self.info.settings.compiler.version), minimum_version):
-            raise ConanInvalidConfiguration(
-                "{} requires C++14, which your compiler does not support.".format(self.name)
-            )
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++14, which your compiler does not support.")
```
Comment (prince-chrismc) [inline recipes/g3log/all/conanfile.py:98]: This prints the wrong cppstd FYI

Comment (prince-chrismc) [reply recipes/g3log/all/conanfile.py:98]:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.")
```

**Review by prince-chrismc — COMMENTED** [review]
Super tiny nit to bot confuse consumers

File: recipes/g3log/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/g3log/all/conanfile.py:98]: @prince-chrismc 
Thanks a lot!

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**